clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerUtils.java

```diff
@@ -211,4 +226,17 @@ else if (t instanceof KafkaException)
             throw new TimeoutException(e);
         }
     }
+
+    public static boolean maybeOverrideEnableAutoCommit(ConsumerConfig config) {
```

Comment:
nit: Not your fault but I find the name of the method a little weird because it does not override anything. It just returns whether auto commit should be enabled or not.

Reply:
It has now been changed to override the configuration. LMK if you think that is a good idea, or if I should revert it.

Reply:
@dajac—following up on this—are you OK with this change?